src/components/SchemaRow.tsx

```diff
@@ -8,21 +8,27 @@ import get = require('lodash/get');
 import map = require('lodash/map');
 import size = require('lodash/size');
 
-import { GoToRefHandler, SchemaNodeWithMeta, SchemaTreeListNode } from '../types';
+import { GoToRefHandler, IExtendableRenderers, SchemaNodeWithMeta, SchemaTreeListNode } from '../types';
 import { isCombiner, isRef } from '../utils';
 import { Types } from './';
 
-export interface ISchemaRow {
+export interface ISchemaRow extends IExtendableRenderers {
   node: SchemaTreeListNode;
   rowOptions: IRowRendererOptions;
   onGoToRef?: GoToRefHandler;
+  toggleExpand: () => void;
 }
 
 const ICON_SIZE = 12;
 const ICON_DIMENSION = 20;
-const ROW_OFFSET = 7;
 
-export const SchemaRow: React.FunctionComponent<ISchemaRow> = ({ node, rowOptions, onGoToRef }) => {
+export const SchemaRow: React.FunctionComponent<ISchemaRow> = ({
+  node,
+  rowOptions,
+  onGoToRef,
+  rowRendererRight,
```

Comment:
This is kind of a hack. Ideally, the entire row should be customizable.

Reply:
That's the solution suggested by @marbemac. I personally don't think it's much of a hack, but a hook :) Similarly to how you'd get plugins a space to render buttons. We can make this more versatile of course in the future, but making the row ultra generic wasn't the plan.

Side note: I don't like the idea of making the whole row customizable. I mean I see the merit of that but I don't think that we need so much flexibility.

Reply:
He suggested 2 solutions - a simple one and also a bit more difficult one but also the better one.
Obviously it's by far better in comparison to the previous solution but having a possibility to customise the entire row would be make the solution more flexible. 
This is just some sort of workaround. I'm fine with that, don't worry, but would really wish it was more flexible. Tree list also exposed such a simple renderer prop initially, but we ended up with the entire row being customizable. The same will likely to apply here at some point, but it's good as is now.

Reply:
Yup, if we need more flexibility I'm all for it and can even do the refactoring myself if help's needed. I think what @marbemac suggested is to go with the simpler suggestion now and I put my bet on it since I think it makes sense for now. Anyway, looks like we're on alignment.

Reply:
Yeah, it's totally okay for now :)

Reply:
I know it's merged, but a quick follow on here (for later PR consideration etc). Apologies if this was already the idea, but:

- Could add the `rowRenderer(props, opts: { DefaultRow: ReactElement })` prop.
- Extract the current elements in the row into a standalone component, expose that as that  `DefaultRow` property passed to rowRenderer opts.
- Up to user of rowRenderer to render `<DefaultRow {...props} />` somewhere if the want it, and then can tweak the props to it (or just pass them through as is), and put whatever they want around it.

Shouldn't be too difficult. For future consideration.

Reply:
Yessir! This is *exactly* what I wanted to see being implemented here and what I suggested in the previous PR. 
I'll do it once I have a bit of spare time